### PR TITLE
added spk for SickRage

### PR DESCRIPTION
**Pull request by cytec**
took a while but i finally managed to make a PR...

### CODE REVIEW

File: spk/sickrage/src/sickrage.sc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[sickbeard-custom]
```
Comment (Diaoul) [inline spk/sickrage/src/sickrage.sc]: sickrage here

File: spk/sickrage/src/wizard/install_uifile [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[{
+    "step_title": "Choose your fork",
+    "items": [{
+        "type": "textfield",
+        "desc": "Configure the fork to use",
```
Comment (Diaoul) [inline spk/sickrage/src/wizard/install_uifile]: Not sure there are a lot of forks out there, maybe add something like "skip this to install default sickrage"

File: spk/sickrage/src/wizard/install_uifile_enu [hunk 3]
```diff
@@ -0,0 +1,22 @@
+[{
+    "step_title": "Choose your fork",
+    "items": [{
+        "type": "textfield",
+        "desc": "Configure the fork to use",
+        "subitems": [{
+            "key": "wizard_fork_url",
+            "desc": "Fork URL",
+            "defaultValue": "git://github.com/cytec/SickRage.git",
```
Comment (Diaoul) [inline spk/sickrage/src/wizard/install_uifile_enu]: why cytec here and SiCKRAGETV elsewhere?

File: spk/sickrage/src/wizard/install_uifile_fre [hunk 4]
```diff
@@ -0,0 +1,22 @@
+[{
+    "step_title": "Choisissez votre fork",
+    "items": [{
+        "type": "textfield",
+        "desc": "Paramètres du fork à utiliser",
+        "subitems": [{
+            "key": "wizard_fork_url",
+            "desc": "URL du fork",
+            "defaultValue": "git://github.com/cytec/SickRage.git",
```
Comment (Diaoul) [inline spk/sickrage/src/wizard/install_uifile_fre]: Same

File: spk/sickrage/src/dsm-control.sh [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+# Others
+INSTALL_DIR="/usr/local/${PACKAGE}"
+PYTHON_DIR="/usr/local/python"
+GIT_DIR="/usr/local/git"
+PATH="${INSTALL_DIR}/bin:${INSTALL_DIR}/env/bin:${PYTHON_DIR}/bin:${GIT_DIR}/bin:${PATH}"
+USER="sickrage"
+PYTHON="${INSTALL_DIR}/env/bin/python"
+GIT="${GIT_DIR}/bin/git"
+SICKBEARD="${INSTALL_DIR}/var/SickRage/SickBeard.py"
```
Comment (Diaoul) [inline spk/sickrage/src/dsm-control.sh]: Shouldn't that be SICKRAGE?

File: spk/sickrage/src/wizard/install_uifile_enu (hunk 3 above)
Comment (cytec) [inline spk/sickrage/src/wizard/install_uifile_enu]: missed those files...

**Comment by cytec** [comment]
@Diaoul updated, i didn't change the french wizzard file as i don't speek french... should i delete it or leave it as it is?

**Comment by Diaoul** [comment]
Add " (suivant pour installer le sickrage par défaut)"

**Comment by neoatomic** [comment]
@cytec Thanks for adding. :)

For Dutch :

```
install_uifile_nld
```

```
[{
    "step_title": "Installatie SickRage",
    "items": [{
        "type": "textfield",
        "desc": "Klik op volgende om SickRage te installeren. (Of vul een alternative URL in.)",
        "subitems": [{
            "key": "wizard_fork_url",
            "desc": "Fork URL",
            "defaultValue": "git://github.com/SiCKRAGETV/SickRage.git",
            "validator": {
                "allowBlank": false
            }
        }, {
            "key": "wizard_fork_branch",
            "desc": "Fork branch",
            "defaultValue": "master",
            "validator": {
                "allowBlank": false
            }
        ]
    }]
}]
```

File: spk/sickrage/Makefile [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+SPK_VERS = $(shell date +%Y%m%d)
+SPK_REV = 1
+SPK_ICON = src/${SPK_NAME}.png
+DSM_UI_DIR = app
+
+DEPENDS  =
+SPK_DEPENDS = "python>2.7.3-3:git"
+
+MAINTAINER = SynoCommunity
+DESCRIPTION = Video File Manager for TV Shows, It watches for new episodes of your favorite shows and when they are posted it does its magic.
```
Comment (Dr-Bean) [inline spk/sickrage/Makefile]: Use a `.` instead of a `,` there: [..]TV Shows. It watches[..]

File: spk/sickrage/src/dsm-control.sh [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+GIT="${GIT_DIR}/bin/git"
+SICKRAGE="${INSTALL_DIR}/var/SickRage/SickBeard.py"
+CFG_FILE="${INSTALL_DIR}/var/config.ini"
+PID_FILE="${INSTALL_DIR}/var/sickrage.pid"
+LOG_FILE="${INSTALL_DIR}/var/Logs/sickrage.log"
+
+
+start_daemon ()
+{
+    su - ${USER} -c "HOME=${INSTALL_DIR}/var PATH=${PATH} ${PYTHON} ${SICKRAGE} --daemon --pidfile ${PID_FILE} --config ${CFG_FILE} --datadir ${INSTALL_DIR}/var/"
```
Comment (Dr-Bean) [inline spk/sickrage/src/dsm-control.sh]: Remove the `-` after `su`.

File: spk/sickrage/src/wizard/install_uifile_nld [hunk 8]
```diff
@@ -0,0 +1,22 @@
+[{
+    "step_title": "Installatie SickRage",
+    "items": [{
+        "type": "textfield",
+        "desc": "Klik op volgende om SickRage te installeren. (Of vul een alternative URL in.)",
```
Comment (Dr-Bean) [inline spk/sickrage/src/wizard/install_uifile_nld]: `"desc": "Klik op volgende om SickRage te installeren. (Of vul een alternatieve URL in.)",`

**Comment by Dr-Bean** [comment]
How about using an `sc-sickrage` group here, instead of the `users` group, similar to `sc-syncthing`? /cc @Diaoul.
(note: we might want to put that in the wizard/set up a FAQ entry on that subject)

**Comment by cytec** [comment]
@Dr-Bean i've fixed your comments. about that `sc-sickrage` group. i can add that but i don't know if that's the best solution because you'll end up with like hundreds of package-specific groups... for stuff like sickrage, sickbeard, sabnzb, transmission etc how about adding a `sc-downloaders` group or something like that? so that we can group a few packages together... most likely if you'll provied sabnzb access to your directory you also want to provide access for sickbeard i think? any thoughts?

**Comment by Dr-Bean** [comment]
I agree with that point of view, in fact, I've brought it up before, here: https://github.com/SynoCommunity/spksrc/issues/1031#issuecomment-46164982.
Maybe split off the discussion in a separate issue (although I think we have an open issue discussing the exact same thing), but I brought it up here because if there's ever a time to easily implement it, it's before the package is published...

**Comment by Dr-Bean** [comment]
@cytec Any idea if https://github.com/SynoCommunity/spksrc/issues/775 is still something we need to concern ourselves with?

**Comment by Diaoul** [comment]
I prefer fine grained permissions but we're using groups (not users) so maybe that makes sense to have a more generic "group" for similar applications.
I bring this issue to Synology in september last year but haven't got any real feedback on that issue since despite my numerous follow ups. I'll tell you if I have any news on that side.

**Comment by Dr-Bean** [comment]
Well, then I vote we go ahead with implementing the group approach. We can revisit the approach when Synology chooses to implement the application user solution (who knows how long that might be...and it probably won't have any form of backward compatibility). In the meantime, the group approach should lessen the issues people are having with setting permissions.

So, as per the proposal @Diaoul mentioned in the other issue: 
- Use `sc_download` for downloaders such as SABnzbd, transmission, rutorrent, etc
- Use `sc_media` for Sickbeard(-derivatives), CouchPotato, Sonarr, etc.
- Unrelated packages that only cover their own setup: one group per package (e.g. Syncthing's approach)

Some details might not have been worked out yet, e.g. we need to make sure that the group is not removed on uninstall, unless it's empty etc. @cytec, are you up for that?

**Comment by Diaoul** [comment]
We use `-` for [Syncthing](https://github.com/SynoCommunity/spksrc/blob/master/spk/syncthing/src/installer.sh#L17) so let's stick to that.

**Comment by cytec** [comment]
@Dr-Bean yup i'll take a look at that when i get some time. As @Diaoul said id stick to `-` for the group names as syncthing currently uses `sc-syncthing`...

also if i remember correctly `syncthing` already checks if the group is empty on uninstall and only removes the group if no users are left

**Comment by gkroone** [comment]
a bit offtoopic, but while discussing group permissions here, I'd like to make another suggestion to give some thought.

I see multiple packages now adding support for USB serial devices, and each package handles the permission setting of these devices on it's own, but all I've seen so far is that they set the permission to 666 or even 777?? for /dev/ttyUSB*, which seems to me like it's insecure. 

I know Ubuntu has a default group called "dialout" which gets added to all the serial ports, and users needing access to such ports get added to group "dialout". woudl it be possible and feasible to do something similar on DSM?

**Comment by Dr-Bean** [comment]
Oh right, the underscore was an honest mistake, I didn't even notice we switched. Dash is fine obviously.
@cytec, looks good to me. The description: I don't have anything better, so let's start with that. We can always change it later.
@gkroone, could you open a new issue for that, where you point out the applicable packages? It's offtopic for this PR, so tracking progress for it might get lost.

**Comment by cytec** [comment]
@Dr-Bean yeah i've renamed the functions to `syno_group_create` and `syno_group_remove` i think that describes pretty good what it does, and one can simply copy paste the functions over to new SPK's and just change the `SYNO_GROUP` and `SYNO_GROUP_DESC` values

File: spk/sickrage/src/installer.sh [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+SERVICETOOL="/usr/syno/bin/servicetool"
+FWPORTS="/var/packages/${PACKAGE}/scripts/${PACKAGE}.sc"
+
+SYNO_GROUP="sc-media"
+SYNO_GROUP_DESC="SynoCommunity's media related group"
+
+syno_group_create ()
+{
+    # Create sync group (Does nothing when sync group already exists)
```
Comment (Diaoul) [inline spk/sickrage/src/installer.sh]: That's not sync group here ;)

**Comment by Diaoul** [comment]
A few changes in the comments I suggest you standardize and remove any reference to the name of the group in the comments so it's "portable".

File: spk/sickrage/src/installer.sh (hunk 9 above)
Comment (cytec) [inline spk/sickrage/src/installer.sh]: damn, thought i got all of those changed to syno group...

**Comment by cytec** [comment]
@Diaoul fixed

**Comment by Dr-Bean** [comment]
Should be good to go, right?

**Comment by Dr-Bean** [comment]
..and to be complete, `sickrage 20150526-1` is published by cytec. Missing a beta tag, but I guess it's too late now ;)

**Comment by neoatomic** [comment]
@Dr-Bean Can confirm it works fine. :)
(However had one small problem, i forget that i already had a user called "sickrage" on the Synology. So got permission errors after the install, but removing the account and reinstalling the spk solved that.)

Thanks to all who made this possible.

**Comment by Dr-Bean** [comment]
@neoatomic Thanks for the feedback, sounds good. Not that we were expecting a lot of issues anyway, plenty of experience with the current Sickbeard packages.

We should probably keep in mind to add a check if users exist and halt the package installation if so. Especially once Synology adds application users it might get more interesting to add that in.

**Comment by neoatomic** [comment]
@cytec @Dr-Bean Could we change the below line: 
https://github.com/SynoCommunity/spksrc/blob/master/spk/sickrage/src/installer.sh#L68 to
`${GIT} clone --recursive -q -b ${wizard_fork_branch:=master} ${wizard_fork_url:=git://github.com/SiCKRAGETV/SickRage.git} ${INSTALL_DIR}/var/SickRage`

The ` --recursive` will allow the download of submodules that where implemented in SickRage. (added NZBtoMedia Repo in the contrub folder.)

**Comment by Dr-Bean** [comment]
Sure that should work for new installs (and when we release a package update. Will you open a PR?

Seeing the linked discussion, you'd probably need a `git submodule update --init --recursive` or something to that effect for future SR updates.

**Comment by Dr-Bean** [comment]
@neoatomic I saw your edit. Note that if the solution still includes working with submodules, we probably will still need to use --recursive, otherwise new installations won't init the submodules. Or will you force a submodule init the first time SR runs? 
Anyway, let us know if we need to make changes.

**Comment by neoatomic** [comment]
Thought we had a solution by using --recursive in the updater but it doesn't seem to work (at least for me) on my Synology. So it still might be necessary to add it to the spk/installer.
Will update this topic as soon as we know more.

**Comment by Dr-Bean** [comment]
Please open a new issue then, that makes the discussion easier to find :)

Fwiw, did a quick test just now (stop SR, `git reset --hard some_old_commit`, verify versionChecker is using `submodule update --init --recursive --force`, remove versionChecker.pyc, start SR, trigger manual update), and I've got content in `contrib/nzbToMedia`.

But, as I said, unless SR forces `submodule update --init` on the very first run of SR, we'll still need to use `--recursive` to cater for people who start off with a fresh install of the package (and that applies to other packagers too, not just SynoCommunity).

**Pull request merged**